### PR TITLE
Fix remote reserve check with inbound claims-in-flight

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Based on #314, this fixes a bug found by chanmon_fail_consistency (and another bug that was introduced in the first fix). See comments/commits for more info.

### CODE REVIEW

File: src/ln/functional_tests.rs [hunk 1]
```diff
@@ -1449,6 +1449,153 @@ fn channel_reserve_test() {
...
+	//  * deliver A's response CS and RAA.
+	//    This results in A having the second HTLC in AwaitingRemovedRemoteRevoke, but B having
+	//    removed it fully. B now has the push_msat plus the first two HTLCs in value.
+	//  * Now B happily sends another HTLC, potentially violating its reserve value from A's point
+	//    of view (if A counts the AwaitingRemovedRemoteRevoke HTLC).
+	let mut nodes = create_network(2);
+	let chan_1 = create_announced_chan_between_nodes(&nodes, 0, 1);
+
+	let b_chan_values = get_channel_value_stat!(nodes[1], chan_1.2);
+	// Route the first two HTLCs. Note that the test is to check that, when you consider the values
```
Comment (ariard) [inline src/ln/functional_tests.rs]: Well IMO this note is simply explaining the test better than the long description above, maybe it should be in top-level comment

File: src/ln/channel.rs [hunk 2]
```diff
@@ -1594,7 +1594,22 @@ impl Channel {
 		// Check our_channel_reserve_satoshis (we're getting paid, so they have to at least meet
 		// the reserve_satoshis we told them to always have as direct payment so that they lose
 		// something if we punish them for broadcasting an old state).
-		if htlc_inbound_value_msat + msg.amount_msat + self.value_to_self_msat > (self.channel_value_satoshis - Channel::get_our_channel_reserve_satoshis(self.channel_value_satoshis)) * 1000 {
+		// Note that we don't really care about our commitment transactions, as the purpose of the
+		// channel reserve is to ensure we can punish *them* if they misbehave, so we ignore any
+		// outbound HTLCs which will not be present in the next commitment transaction we send
+		// them (at least for fulfilled ones, failed ones won't modify value_to_self).
+		// Note that we will send HTLCs which we'd think violate the reserve value if we do not do
```
[the remark this answers is not in the document]
Comment (TheBlueMatt) [reply src/ln/channel.rs]: This is a bit confusingly worded, I'll rephrase it to "Note that we will send HTLCs which another instance of rust-lightning would think violate the reserve value..."

File: src/ln/channel.rs [hunk 3]
```diff
@@ -1594,7 +1594,22 @@ impl Channel {
 		// Check our_channel_reserve_satoshis (we're getting paid, so they have to at least meet
 		// the reserve_satoshis we told them to always have as direct payment so that they lose
 		// something if we punish them for broadcasting an old state).
-		if htlc_inbound_value_msat + msg.amount_msat + self.value_to_self_msat > (self.channel_value_satoshis - Channel::get_our_channel_reserve_satoshis(self.channel_value_satoshis)) * 1000 {
+		// Note that we don't really care about our commitment transactions, as the purpose of the
+		// channel reserve is to ensure we can punish *them* if they misbehave, so we ignore any
```
[the remark this answers is not in the document]
Comment (TheBlueMatt) [reply src/ln/channel.rs]: Sure.

File: src/ln/channel.rs [hunk 4]
```diff
@@ -1594,7 +1594,22 @@ impl Channel {
 		// Check our_channel_reserve_satoshis (we're getting paid, so they have to at least meet
 		// the reserve_satoshis we told them to always have as direct payment so that they lose
 		// something if we punish them for broadcasting an old state).
-		if htlc_inbound_value_msat + msg.amount_msat + self.value_to_self_msat > (self.channel_value_satoshis - Channel::get_our_channel_reserve_satoshis(self.channel_value_satoshis)) * 1000 {
+		// Note that we don't really care about our commitment transactions, as the purpose of the
```
[the remark this answers is not in the document]
Comment (TheBlueMatt) [reply src/ln/channel.rs]: Hmm, I'll rephrase, but what I really meant is that we dont care about lack of to_remote output in our local commitment txn as we werent gonna use it to punish them anyway.

File: src/ln/functional_tests.rs (hunk 1 above)
Comment (TheBlueMatt) [reply src/ln/functional_tests.rs]: Done.

**Comment by TheBlueMatt** [comment]
Can you confirm the new comments are clearer-enough?

File: src/ln/channel.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ariard) [reply src/ln/channel.rs]: Speaking of rust-lightning instance means this point isn't cleary specified in specs right ? Maybe think to open an issue on this.

File: src/ln/channel.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ariard) [reply src/ln/channel.rs]: Ah better!

**Comment by TheBlueMatt** [comment]
I didn't actually double-check the spec (should do to make sure it's clear) but this is the only interpretation that makes sense, I think.

**Comment by ariard** [comment]
Good, read again every litigious comment, get them better
